src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopyRowSetTest.java

```diff
@@ -0,0 +1,68 @@
+package com.microsoft.sqlserver.jdbc.bulkCopy;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import javax.sql.RowSetMetaData;
+import javax.sql.rowset.CachedRowSet;
+import javax.sql.rowset.RowSetFactory;
+import javax.sql.rowset.RowSetMetaDataImpl;
+import javax.sql.rowset.RowSetProvider;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opentest4j.TestAbortedException;
+
+import com.microsoft.sqlserver.jdbc.RandomData;
+import com.microsoft.sqlserver.jdbc.RandomUtil;
+import com.microsoft.sqlserver.jdbc.SQLServerBulkCopy;
+import com.microsoft.sqlserver.jdbc.SQLServerStatement;
+import com.microsoft.sqlserver.jdbc.TestUtils;
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+
+public class BulkCopyRowSetTest extends AbstractTest {
+    
+    private static String tableName = AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("BulkCopyFloatTest"));
+    
+    @Test
+    public void testBulkCopyFloatRowSet() throws SQLException {
+        try (Connection con = DriverManager.getConnection(connectionString)) {
+            RowSetFactory rsf = RowSetProvider.newFactory();
+            CachedRowSet crs = rsf.createCachedRowSet();
+            RowSetMetaData rsmd = new RowSetMetaDataImpl();
+            rsmd.setColumnCount(1);
+            rsmd.setColumnName(1, "c1");
+            rsmd.setColumnType(1, java.sql.Types.FLOAT);
+            
+            crs.setMetaData(rsmd);
+            crs.setFloat(1, RandomData.generateReal(false));
+            
+            try (SQLServerBulkCopy bcOperation = new SQLServerBulkCopy(con)) {
+                bcOperation.setDestinationTableName(tableName);
+                bcOperation.writeToServer(crs);
+            }

```

Comment:
Must also do verification to check if the fetched data is same as inserted.
Consider generating multiple Random Value(s) (different range of data) during setup (beforeAll) and during insert also add additional IDENTITY column to table to be able to fetch in the same order.

Reply:
Added check. I didn't feel like running the same test an arbitrary number of times was necessary though.

Reply:
What I meant to say was there are 2 types of Floats we should test (least of all) with null value
`RandomData.generateFloat(24, false);`
`RandomData.generateFloat(53, false);`
`RandomData.generateFloat(53, true); //null`

Reply:
As for the null case, it seems we can't put nulls when using bulk copy using a RowSet because the CachedRowSetImpl doesn't allow null values.
As for the generateFloat(24) and generateFloat(53), they're actually an identical case in this scenario. When the user specifies `FLOAT` in the RowSetMetaData, the driver will send the java equivalent of a float, which truncates the 53 case down to 24, except now we have to deal with rounding errors in the assertEquals line.